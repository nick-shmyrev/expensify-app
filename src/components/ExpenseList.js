import React from 'react';
import { connect } from 'react-redux';
import { getVisibleExpenses } from '../store/selectors/expenses-selectors';

import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = props => (
  <div>
    {
      props.expenses.length === 0 ?
        (<p>No expenses to display.</p>) :
        (<ul>
          {props.expenses.map(expense => (
            <ExpenseListItem {...expense} key={expense.id}/>
          ))}
        </ul>)
    }
  </div>
);

const mapStateToProps = state => ({
  expenses: getVisibleExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseList);

