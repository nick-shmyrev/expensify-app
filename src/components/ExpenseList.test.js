import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from './ExpenseList';

describe('ExpenseList component', () => {
  const expenses = [
    {
      id: '1234-abcd',
      description: 'Expense to remove',
      note: 'With a note',
      amount: 13,
      createdAt: 0,
    },
    {
      id: 'ccd-42',
      description: 'Expense to keep',
      note: 'With a note',
      amount: 42,
      createdAt: 0,
    },
    {
      id: 'ccd-43',
      description: 'Expense to keep too',
      note: 'With a note',
      amount: 42,
      createdAt: 0,
    },
  ];
  
  test('should render with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should correctly render without expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
