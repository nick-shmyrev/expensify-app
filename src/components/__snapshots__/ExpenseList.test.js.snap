// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseList component should correctly render without expenses 1`] = `
<div>
  <p>
    No expenses to display.
  </p>
</div>
`;

exports[`ExpenseList component should render with expenses 1`] = `
<div>
  <ul>
    <ExpenseListItem
      amount={13}
      createdAt={0}
      description="Expense to remove"
      id="1234-abcd"
      key="1234-abcd"
      note="With a note"
    />
    <ExpenseListItem
      amount={42}
      createdAt={0}
      description="Expense to keep"
      id="ccd-42"
      key="ccd-42"
      note="With a note"
    />
    <ExpenseListItem
      amount={42}
      createdAt={0}
      description="Expense to keep too"
      id="ccd-43"
      key="ccd-43"
      note="With a note"
    />
  </ul>
</div>
`;
